cmd/rekor-cli/app/log_info.go

```diff
@@ -115,19 +142,24 @@ func verifyTree(rekorClient *rclient.Rekor, signedTreeHead, serverURL string) er
 		return errors.New("signature on tree head did not verify")
 	}
 
-	if err := proveConsistency(rekorClient, oldState, sth); err != nil {
+	if err := proveConsistency(rekorClient, oldState, sth, treeID); err != nil {
 		return err
 	}
 
 	if viper.GetBool("store_tree_state") {
+		if treeID != "" {
+			if err := state.Dump(treeID, &sth); err != nil {
+				log.CliLogger.Infof("Unable to store previous state: %v", err)
+			}
+		}
 		if err := state.Dump(serverURL, &sth); err != nil {
```

Comment:
Some state will get stored twice! But when reading from the file the client will always prefer the `treeID` over the URL.

Reply:
Okay cool, yeah I see that in `oldState`. Just trying to understand the mechanism here - the URL covers all trees, right? What does dumping by `treeID` (if available) add?

Reply:
So if we have two shards, they'll be accessed at the same URL (e.g. rekor.sigstore.dev). We can't store state for both of them, because the key by URL would be the same. The TreeID is unique for each shard, so we can store state for all shards at once.